adapters/nft-key/index.js

```diff
@@ -0,0 +1,164 @@
+        this.range = 500;
+        this.chunkSize = 6;
+        this.sdk = new Avalanche(this);
+    }
+
+    run = async () => {
+        const s = await this.getSymbol();
+        this.sdk = this.loadSdk();
+        this.symbol = s;
+        await this.sdk.run();
+    };
+
+    loadSdk = () => {
+        return new Avalanche(this);
+    };
+    getSymbol = async () => {
+        const resp = await axios.get(
+            `${URL}/token-metadata?key=${KEY}&token_address=${this.token}&protocol=${this.protocol}`,
+            {
+                headers: {
+                    "User-Agent":
+                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4619.141 Safari/537.36",
+                },
+            },
+        );
+        const symbol = resp.data;
+        return symbol;
+    };
+    getPrice = async timestamp => {
+        const resp = await axios.get(
+            `${URL}/token-price?key=${KEY}&token_address=${this.token}&protocol=${this.protocol}&timestamp=${timestamp}`,
+            {
+                headers: {
+                    "User-Agent":
+                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4619.141 Safari/537.36",
+                },
+            },
+        );
+        return resp.data;
+    };
+    stop = async () => {
+        this.sdk.stop();
+    };
+
+    getBuyer = async event => {
+        const buyer = event.returnValues.owner;
+        if (event.event === "Offer, OfferCreated, OfferCancelled, OfferAccepted") {
+            const txReceipt = await this.sdk.getTransactionReceipt(event.transactionHash);
+            if (txReceipt === null) {
+                return null;
+            }
+            return txReceipt.from;
+        }
+        return buyer;
+    };
+
+    process = async event => {
+        const block = await this.sdk.getBlock(event.blockNumber);
+        const timestamp = moment.unix(block.timestamp).utc();
+        const po = await this.getPrice(block.timestamp);
+        const nativePrice = new BigNumber(event.returnValues.cost).dividedBy(10 ** this.symbol.decimals);
+        const buyer = await this.getBuyer(event);
+        if (!buyer) {
+            return;
+        }
+
+        const labelHash = event.returnValues.label;
+        const tokenId = new BigNumber(labelHash, 16).toFixed();
+        const entity = {
+            provider_name: this.name, // the name of the folder
+            provider_contract: this.contract, // the providers contract from which you get data
+            protocol: this.protocol,
+            nft_contract: this.contract,
+            nft_id: tokenId,
+            token: this.token,
+            token_symbol: this.symbol.symbol,
+            amount: 1,
+            price: nativePrice.toNumber(),
+            price_usd: nativePrice.multipliedBy(po.price).toNumber(),
+            seller: this.contract, // Exchange happens on NFT Key
```

Comment:
I'm pretty sure NFT Key marketplace is not a seller.

Reply:
They provide 2 different sets of Events Emitted in their Docs...perhaps the Global Marketplace is the correct set to use I would think then. 

This is where I found the Events:
https://docs.nftkey.app/faqs/